### ISSUE TITLE
Consider themeing other libraries

### ISSUE THREAD

**Issue by alexanderchan** [SAP/fundamental-react#207]
A bit of a rant here but I wanted to explore trying to re-use the best components out there.

Instead of trying and going through the entire build and bugfixing process, would it be possible to take the best off the shelf components and theme them with fiori fundamentals?  Why re-invent the datepicker that somone like [airbnb](https://github.com/airbnb/react-dates) has already built and bug tested when we can liberally re-use and contribute back to a datepicker that is literally used by millions of people?

To me it's more value if SAP chooses the best components, themes them and then makes them available under one library than to spend that effort to re-build things where we have to find all sorts of bugs because we think we can build a better version.

AirBnb's datepicker for example has already 512 closed PRs and 600 issues closed and more than likely has 95% of the functionality we would need.  I'm not saying that SAP can't do it better, but perhaps starting with their base is an option. *insert standing on shoulders of giants quote here*

From my experience as someone who's had to re-use a datepicker library and built an autocomplete input, I know there are countless edge cases around using these components that go beyond just "pick a date" and "type ahead".  Some of the keyboard shortcuts, internationalization, date ranges, default start, end, and usability testing all have to be implemented and we are starting a few years behind the other libraries. 

Why not just take their open source code where applicable?  

The downsides are:

1) as maintainers you have to know and contribute to the chosen library for any features or fork if they are not accepted
2) spend time to theme

Upsides are

1) Most of the code, bugs, cross-browser/cross-platform has already been tested and the paid for by the source team and contributors 

I'm curious about any thoughts about "base" vs "build".  Similar to the buy vs build argument but there is no actual price for the open source.  We use sap because it's well integrated -- Can we integrate the best off the shelf open source products?

In a react-app there's nothing stopping us from importing another library ourselves of course.  The slowdown for customer projects would be that they would prefer it to be themed the same.  Only developers care about the API.  End users only see the control.  Developers are delighted when it's easy to delight our customers and I wonder if there's some way we can easily re-use the best open source out there?

**Comment by bcullman** [comment]
The goal of Fundamental is sort of revolutionary - components are described as HTML and CSS alone through a reference design in https://github.com/SAP/fundamental.

It is then up to Fundamental-react, fundamental-ngx, fundamental-vue (of fundamental-x) to emit that precise HTML so that the out-of-the box standard CSS provided by https://github.com/SAP/fundamental can properly style it.

Our Aim is to bring together applications built by different teams on different platforms (UI5, fundamental-react, fundamental-vue, fundamental-ngx) and have the experience feel truly seem-less to the end user.

AirBnB's react datepicker is just that - a datepicker for react alone.  What about a date picker for fundamental-ngx, fundamental-vue? Choosing to skin an existing datepicker library each of them puts us at the mercy of the design of each - *each* with their own HTML output, css classnames, and behaviour quirks. 

In short, there would be no way to absolutely *guarantee* that components from each fundemental-x library would  be consistent with each other.

Of course, that does not mean that we are not taking advantage of open sources libraries - here is a testament to that:

 https://github.com/SAP/fundamental-react/blob/master/package.json#L34-L106

**Comment by jbadan** [comment]
Due to the constraints of being attached to the fundamental-styles repository for html markup and css, this isn't possible.
